library/hawkular_alerts_group.py

```diff
@@ -89,42 +89,90 @@
 
 
 class HawkularAlertsGroupTrigger(object):
-    """ Hawkular Alerts object to create group triggers in Hawkular
+    """ Hawkular Alerts object to create and update group triggers in Hawkular
     """
     def __init__(self, module, tenant, hostname, port, scheme, token, context):
         self.module  = module
         self.client  = hawkular.alerts.HawkularAlertsClient(tenant, host=hostname, port=port, scheme=scheme, token=token, context=context)
         self.changed = False
 
-    def create_group_trigger(self, name, group_id, severity, enabled):
-        """ Creates a group trigger in Hawkular Alerting component
+    def required_updates(self, trigger, name, severity, enabled):
+        """ Checks whether an update is required for the group trigger
+
+            Returns:
+                True if the passed trigger name, severity or enabled status
+                are different from the ones set in the trigger, False otherwise
+        """
+        updates = {}
+        if (name and trigger.name != name):
+            updates["name"] = name
+        if (severity and trigger.severity != severity):
+            updates["severity"] = severity.upper()
```

Comment:
I think this should use `getattr(hawkular.alerts.Severity, severity.upper())`.

Or better, do that translation at higher level — in `create_or_update_group_trigger()` or even `main()` and then create/updating methods could just assign/compare severities as-is.